Libraries/DirectMLX.h

```diff
@@ -2004,7 +2004,11 @@ namespace dml
     // ---------------------------------------------------------------------------------------------------------------
 
     // If `axes` is not specified, by default this reduces the entire tensor to single element.
-    inline Expression Reduce(Expression input, DML_REDUCE_FUNCTION function, Span<const uint32_t> axes = {})
+    inline Expression Reduce(
+        Expression input, 
+        DML_REDUCE_FUNCTION function, 
+        DML_TENSOR_DATA_TYPE targetDataType = DML_TENSOR_DATA_TYPE_UNKNOWN, 
```

Comment:
[naming] Would it be more consistent to just name this `outputDataType`, like other places?

e.g. `Expression ElementWiseUnary(Expression input, DML_TENSOR_DATA_TYPE outputDataType = DML_TENSOR_DATA_TYPE_UNKNOWN)`

I don't think we need both an output data type *and* a target data type.